codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/rustlang/WritableTest.kt

```diff
@@ -75,4 +75,36 @@ internal class RustTypeParametersTest {
 
         writer.toString() shouldContain "'<()>'"
     }
+
```

Comment:
there are no tests of the ambiguous behavior when `join` is called with no args.

IMO, I'd remove the default arguments, I think they just will make things confusing

Reply:
The default separator was added to mirror https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html. Will remove it.